codes/classical/bits/IPP.yml

```diff
@@ -0,0 +1,30 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+
+
+code_id: ipp
+physical: bits
+logical: bits
+
+
+name: 'Code with Identifiable Parent Property'
+introduced: '\cite{//doi.org/10.1137/S0895480100376848}'
+
+
+description: 'If \(C\) is a \(q\)-ary code of length \(n\) and \(a\) and \(b\) are two codewords, then \(c\) is called a descendant of \(a\) and \(b\) if \(c_i \in \{a_i,b_i\}\), for \(i \in \{1,2,...,n\}\).Code \(C\) with the property that given any descendant \(c\) one can always identify at least one of the parent codewords in \(C\)'
+
+
+notes:
+  - 'For a \(q\)-ary code of length \(n\) with IPP its size is upper bounded by \(3q^{\frac{n}{3}} \) and lower bounded by \(c (\frac{q}{4})^{n/3}\) where \(c\) is a constant.'
+  - 'More papers \cite{//doi.org/10.1137/S0895480100376848}.'
+
+
+relations:
+
```

Comment:
parent: ecc

Reply:
I would not say that ECC is a parent: the IPP codes do not really perform error correction. They are indeed codes, designed for a somewhat different task.

Reply:
Not really, this is a somewhat ad hoc combinatorial question, that comes in many variations, including a few not appearing here, among them
_Separable codes_ (https://doi.org/10.1109/TIT.2011.2174614)
_Separating codes_ https://doi.org/10.1109/T-C.1969.222707) and http://mi.mathnet.ru/eng/ppi228
_Hash codes_ (https://doi.org/10.1109/ISIT.1997.613089)
_Robust IPP codes_ [10.1109/TIT.2012.2220522](https://doi.org/10.1109/TIT.2012.2220522)

Reply:
Would you be able to formulate a loose enough umbrella defn? Copyright-infringement, piracy, fingerprinting, or copyright codes maybe? Can one of the be argued to be a parent of the other ones? I can take a dive at some point if need be.

Reply:
Something like: "Copyright protection, or _digital fingerprinting_ codes are designed to prevent unauthorized distribution of digital contents through assigning individual tamper-proof labels to legal copies." 
To my eyes, this is too far from a mathematical definition to be useful.